File: java/Punto.java

public class Punto {
	
	private static int alloc = 0;
	
	private final int x;
	private final int y;
	
	public Punto() {
		this(0, 0);
	}
	
	public Punto(int x , int y) {
		
		this.x = x;
		this.y = y;
		this.alloc++;
	}
	
	public int getX() {
		return this.x;
	}
	
	public int getY() {
		return this.y;
	}
	
	public String toString() {
		return x + ", " + y;
	}
	
	public static int getAlloc() {
		return alloc;
	}
}

File: cpp/MdP/Punto/Punto.cpp
#include "Punto.hh"
#include<iostream>
#include<cmath>

Punto::Punto(double x,double y) : x(x), y(y){};
Punto::Punto() : x(0), y(0){};

double Punto::distanzaDa(const Punto& p) const {
    double dx = x - p.x;
    double dy = y - p.y;

    return sqrt(dx*dx + dy*dy);
};

void Punto::stampa() const {
    std::cout<<x << ", " << y << std:: endl;
};

double Punto::getX() const {
    return x;
}

double Punto::getY() const {
    
}